gnomad/utils/annotations.py

```diff
@@ -1446,3 +1373,166 @@ def annotate_downsamplings(
     )
 
     return t
+
+
+def compute_freq_by_strata(
+    mt: hl.MatrixTable,
+    strata_expr: List[Dict[str, hl.expr.StringExpression]],
+    downsamplings: Optional[List[int]] = None,
+    ds_pop_counts: Optional[Dict[str, int]] = None,
+    entry_agg_funcs: Optional[Dict[str, Tuple[Callable, Callable]]] = None,
+) -> hl.Table:
+    """
+    Compute call statistics and, when passed, entry aggregation function(s) by strata.
+
+    The computed call statistics are AC, AF, AN, and homozygote_count. Downsamplings are
+    added to the strata when downsamplings when passed. The entry aggregation functions
+    are applied to the MatrixTable entries and aggregated by strata.
+
+    .. note::
+        This function is primarily used through annotate_freq but can be used
+        independently if desired.
+
+    :param mt: Input MatrixTable.
+    :param strata_expr: List of dicts of strata expressions.
+    :param downsamplings: Optional list of downsampling groups.
+    :param ds_pop_counts: Optional dict of population counts for downsampling groups.
+    :param entry_agg_funcs: Optional dict of entry aggregation functions. When
+        specified, additional annotations are added to the output Table/MatrixTable.
+        The keys of the dict are the names of the annotations and the values are tuples
+        of functions. The first function is used to transform the `mt` entries in some
+        way, and the second function is used to aggregate the output from the first
+        function.
+    :return: Table or MatrixTable with allele frequencies by strata.
+    """
```

Comment:
Do you think we need to add errors like we have in the other function? We could also consider adding a `_` to this function name indicating that it's mostly intended to be used internally

Reply:
I added errors but if you think they are not enough coverage, we can mark it as a private function but I would like to keep it open I think